<?xml version="1.0"?>
<launch>
  <arg name="algo" default="AStar" />
  <arg name="c1" default="true" />
  <arg name="c2" default="true" />

  <arg name="s1" default="0.0" />
  <arg name="s2" default="0.0" />
  <arg name="s3" default="0.0" />
  <arg name="s4" default="0.0" />

  <arg name="nnodes" default="300" />
  <arg name="ic" default="3" />
  <arg name="uinterval" default="1.0" />
  <arg name="dis2p" default="0.3" />
  <arg name="dis2o" default="0.85" />
  <arg name="s_angle" default="25.0" />
  <node pkg="path_follower" type="path_follower_node" name="path_follower" output="screen" respawn="true">
    <param name="use_obstacle_map" value="false" />
    <param name="use_vfh" value="false" />
    <param name="use_path_lookout" value="false" />
    
    <param name="controller" value="differential_orthexp" />
    
    <param name="waypoint_tolerance" value="2.0" />
    <param name="goal_tolerance" value="0.4" />
    <param name="max_distance_to_path" value="2.0" />
    <param name="min_velocity" value="0.1" />
    <param name="max_velocity" value="2.0" />
    <param name="collision_box_width" value="0.4" />
    <param name="collision_box_crit_length" value="0.6" />
    <param name="collision_box_min_length" value="1.0" />
    <param name="collision_box_max_length" value="1.5" />
    <param name="collision_box_velocity_factor" value="1.0" />
    <param name="collision_box_velocity_saturation" value="1.0" />
    <param name="obstacle_scale_distance" value="1.0" />
    <param name="obstacle_scale_lifetime" value="5.0" />
    <param name="path_width" value="0.5" />
 
    <param name="kp" value="1.0" />
    <param name="kd" value="1.0" />

    <param name="look_ahead_dist" value="2.0" />
    <param name="k" value="2.0" />
    <param name="k_o" value="0.0" />
    <param name="k_g" value="0.4" />
    <param name="k_w" value="0.75" />
    <param name="k_curv" value="1.0" />

    <!-- local planner -->
    <param name="algo" value="$(arg algo)"/>
    <!-- constraints -->
    <param name="c1" value="$(arg c1)"/>
    <param name="c2" value="$(arg c2)"/>
    <!-- scorers -->
    <param name="s1" value="$(arg s1)"/>
    <param name="s2" value="$(arg s2)"/>
    <param name="s3" value="$(arg s3)"/>
    <param name="s4" value="$(arg s4)"/>
    <!-- planner params-->
    <param name="nnodes" value="$(arg nnodes)" />
    <param name="ic" value="$(arg ic)" />
    <param name="uinterval" value="$(arg uinterval)" />
    <param name="dis2p" value="$(arg dis2p)" />
    <param name="dis2o" value="$(arg dis2o)" />
    <param name="s_angle" value="$(arg s_angle)" />

    <remap from="scan" to="scan/filtered" />
  </node>
  
</launch>
